### PR TITLE
Initial run

### PR DESCRIPTION
**Pull request by st-murray**
Adding lot's of changes

### CODE REVIEW

File: molecule/docker/tests/test_localectl.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Make sure localectl set's LANG= test value.
+def test_localectl_locale(host):
+    host.run('localectl set-locale LANG=test')
```
Comment (greg-hellings) [inline molecule/docker/tests/test_localectl.py]: It looks like your tests are modifying the system, is that right? Wouldn't it be better if lines 3 and 10 were deleted and the assertions only verified that the role properly configures the system?

File: tasks/main.yml [hunk 2]
```diff
@@ -1 +1,36 @@
 # role tasks
+
+- name: Find the current locale setting
+  slurp:
+    src: /etc/locale.conf
```
Comment (greg-hellings) [inline tasks/main.yml:4]: How large are this and the other flies you're slurping likely to be? When you use `slurp` to fetch the entire file, we want to be careful that the files aren't going to grow to a large size and take up a bunch of bandwidth if our need is only for a single line or two in the file.

File: tasks/main.yml [hunk 3]
```diff
@@ -1 +1,36 @@
...
+  register: host_locale
+
+- name: update locale if needed
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+  command: localectl set-locale LANG="{{ localectl_locale}}"
+  when: localectl_locale not in host_locale.content | b64decode
```
Comment (greg-hellings) [inline tasks/main.yml:11]: Are these conditionals sufficiently narrow to avoid problems when likely comments are present in a config file? Do we want to somehow narrow this down?

Tying in with the comment above about how large the resulting file is, we might want to fetch the contents of the file currently by passing the results through a pipe to eliminate lines that are clearly wrong, commented, or otherwise might get in the way of these tests. A simple example might be something like this:

`grep <file> -v -e '^#' | sed -E -e 's/#.*$//' | grep <option name>`

The first grep will eliminate lines that begin with a comment character ("#").
The sed will strip out comments that live within a line.
The last section will return online the line (or lines) that include the name of the setting being searched for.

Running this pipe through the `shell` module in Ansible and registering the result will allow you to run this conditional and others with greater (not perfect, but greater) confidence that you're not slurping extra data across the network and also reduces the chance that a string like "us" that's anywhere in the comment section of a file might cause this role to misbehave.

**Review by greg-hellings — CHANGES_REQUESTED** [review]

File: molecule/docker/tests/test_localectl.py (hunk 1 above)
Comment (st-murray) [reply molecule/docker/tests/test_localectl.py]: Tests have been modified to only verify the role worked correctly

File: tasks/main.yml (hunk 2 above)
Comment (st-murray) [reply tasks/main.yml:4]: The two remaining files that I need to use slurp on are quite small

File: tasks/main.yml (hunk 3 above)
Comment (st-murray) [reply tasks/main.yml:11]: I have updated all the conditionals in the role to be much less risky.

**Comment by st-murray** [comment]
Many updates and changes to the role.  Please review

**Comment by st-murray** [comment]
Please review the latest updates

File: molecule/docker/tests/test_localectl.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+# Make sure localectl_keymap was set correctly.
+def test_localectl_keymap(host):
+    assert host.run('grep KEYMAP /etc/vconsole.conf').stdout == \
+        'KEYMAP=uk'
+
+
+# Make sure localectl_x11_keymap was set correctly.
+def test_localectl_x11_keymap(host):
+    assert host.run("sed -n 's/Option//p' \
```
Comment (tehsmyers) [inline molecule/docker/tests/test_localectl.py:17]: This sed looks fragile; it could use a comment explaining what it's doing and why so that if/when it begins to fail on different systems a future developer does not need to guess at the original author's intent (even in the case where both of those people are the same person, i.e. you).

File: molecule/docker/molecule.yml [hunk 5]
```diff
@@ -32,8 +34,8 @@ verifier:
     v: true
   lint:
     name: flake8
-  # Using the shared directory is useful for sharing tests across scenarios,
-  # but is not a requirement. For scenario specific tests, add the appropriate
-  # file path to the test or test directory below
-  additional_files_or_dirs:
-    - ../shared/*
\ No newline at end of file
+    # Using the shared directory is useful for sharing tests across scenarios,
```
Comment (tehsmyers) [inline molecule/docker/molecule.yml]: Commented out blocks in revision control are effectively redundant, I recommend just deleting this if it won't be used.

File: tasks/main.yml [hunk 6]
```diff
@@ -1 +1,44 @@
...
+- name: Find the current locale setting
+  slurp:
+    src: /etc/locale.conf
+  register: host_locale
+
+- name: Update locale if needed
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+  command: localectl set-locale LANG="{{ localectl_locale }}"
+  when: localectl_locale not in host_locale.content | b64decode
```
Comment (tehsmyers) [inline tasks/main.yml:11]: `localectl set-locale` can take multiple arguments. This will fail if, for example, I use this role to only set numeric representations to a specific locale with `localectl_locale: LC_NUMERIC=some_locale`.

File: tasks/main.yml [hunk 7]
```diff
@@ -1 +1,44 @@
...
+
+# Grep for KEYMAP= and fail if return code empty
+- name: Find the current keymap setting
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+  shell: grep /etc/vconsole.conf -v -e '^#' | sed -E -e 's/#.*$//' | grep
```
Comment (tehsmyers) [inline tasks/main.yml]: The comment above this task doesn't tell the whole story; this does more than just "Grep for KEYMAP=...". The comment should be updated to reflect the work being done and the intent of this task.

**Review by tehsmyers — CHANGES_REQUESTED** [review]

File: molecule/docker/molecule.yml (hunk 5 above)
Comment (st-murray) [reply molecule/docker/molecule.yml]: Commented out block has been removed

File: tasks/main.yml (hunk 7 above)
Comment (st-murray) [reply tasks/main.yml]: Comment has been updated

File: molecule/docker/tests/test_localectl.py (hunk 4 above)
Comment (st-murray) [reply molecule/docker/tests/test_localectl.py:17]: Comments have been added as to my intent on the sed command.

File: tasks/main.yml (hunk 6 above)
Comment (st-murray) [reply tasks/main.yml:11]: I have updated the README with a disclaimer that this part of the role is only intended to set the LANG or language designation of the system.

**Comment by st-murray** [comment]
Another round of edits for review please

File: molecule/openstack/molecule.yml [hunk 8]
```diff
@@ -16,7 +18,7 @@ provisioner:
     name: ansible-lint
   playbooks:
     prepare: ../shared/prepare.yml
-    converge: ../shared/playbook.yml
+    converge: ../docker/playbook.yml
```
Comment (greg-hellings) [inline molecule/openstack/molecule.yml]: Why did you move these out of the `shared` folder, when they're still shared?

File: defaults/main.yml [hunk 9]
```diff
@@ -1 +1,20 @@
-# defaults file for localectl
+# defaults file for localectl.
+
+# Set your systems language designation.
+# Example
+#localectl_locale: en_US.UTF-8
```
Comment (greg-hellings) [inline defaults/main.yml]: Why do we have defaults that are commented out? If this is meant for purely documentation purposes, then these values should be in `README.md` and not here. If they're actual defaults, then they should not be commented out here.

**Review by greg-hellings — CHANGES_REQUESTED** [review]

File: molecule/openstack/molecule.yml (hunk 8 above)
Comment (st-murray) [reply molecule/openstack/molecule.yml]: Moved the playbook back to shared directory and updated molecule.yml file.

File: defaults/main.yml (hunk 9 above)
Comment (st-murray) [reply defaults/main.yml]: I have removed them from defaults and updated the README.

**Comment by st-murray** [comment]
Another round of changes for review please.   updated defaults file, README and tasks/main.yml to address new ansible lint issue.   https://github.com/ansible/ansible-lint/issues/442
